react/src/components/PostHogFeature.tsx

```diff
@@ -0,0 +1,94 @@
+import { useFeatureFlagPayload, useFeatureFlagVariantKey, usePostHog } from '../hooks'
+import React, { RefObject, useEffect, useMemo, useRef, useState } from 'react'
+import { PostHog } from '../context'
+
+export type PostHogFeatureProps = {
+    flag: string
+    children: React.ReactNode | ((payload: any) => React.ReactNode)
+    match?: string | boolean
+    visibilityObserverOptions?: IntersectionObserverInit
+}
+
+export function PostHogFeature({
+    flag,
+    match,
+    children,
+    visibilityObserverOptions,
+}: PostHogFeatureProps): JSX.Element | null {
+    const posthog = usePostHog()
+    const payload = useFeatureFlagPayload(flag)
+    const variant = useFeatureFlagVariantKey(flag)
+    const [clickTracked, setclickTracked] = useState(false)
+
+    if (match === undefined || variant === match) {
+        const childNode: React.ReactNode = typeof children === 'function' ? children(payload) : children
+        return (
+            <div
+                onClick={() => {
+                    if (!clickTracked) {
+                        trackClicks(flag, posthog)
+                        setclickTracked(true)
+                    }
+                }}
+            >
+                <VisibilityTracker flag={flag} options={visibilityObserverOptions}>
+                    {childNode}
+                </VisibilityTracker>
+            </div>
+        )
+    }
+
+    return null
+}
+
+function trackClicks(flag: string, posthog?: PostHog) {
+    posthog?.capture('$feature_flag_clicked', { feature_flag: flag, $set: { [`$feature_interaction/${flag}`]: true } })
+}
+
+function trackVisibility(flag: string, posthog?: PostHog) {
+    posthog?.capture('$feature_flag_viewed', { feature_flag: flag })
```

Comment:
similar confusion here as above, `feature_flag_viewed` means more like `feature_flag_component_viewed` ?

Reply:
Hmm, I'll change these to `$feature_view` and `$feature_interaction`